WordPress/Sniffs/Theme/NoTitleTagSniff.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     https://make.wordpress.org/core/handbook/best-practices/coding-standards/
+ */
+
+/**
+ * WordPress_Sniffs_Theme_NoTitleTagSniff.
+ *
+ * Forbids the use of the <title> tag, unless it is within a <svg> tag.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    carolinan
+ */
+class WordPress_Sniffs_Theme_NoTitleTagSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Property to keep track of whether a <svg> open tag has been encountered.
+	 *
+	 * @var bool
+	 */
+	private $in_svg = false;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_INLINE_HTML, // Finds inline html.
+			T_CONSTANT_ENCAPSED_STRING, // Finds the title tag in php.
+		 );
+	} // end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token in
+	 *                                        the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens  = $phpcsFile->getTokens();
+		$content = $tokens[ $stackPtr ]['content'];
+
+		// No need to check an empty string.
+		if ( '' === trim( $content ) ) {
+			return;
+		}
+
+		// Are we in a <svg> tag ?
+		if ( true === $this->in_svg ) {
+			if ( false === strpos( $content, '</svg>' ) ) {
+				return;
+			} else {
+				// Make sure we check any content on this line after the closing svg tag.
+				$this->in_svg = false;
+				$content      = trim( substr( $content, ( strpos( $content, '</svg>' ) + 6 ) ) );
+			}
+		}
+
+		// We're not in svg, so check if it there's a <svg> open tag on this line.  PHP 5.2 create a seperate token for `<s` we need to check that the first two letters in the next token is vg.
+		if ( false !== strpos( $content, '<svg' ) || ( '<s' === trim( $content ) && 'vg' === substr( $tokens[ ( $stackPtr + 1 ) ]['content'], 0, 2 ) ) ) {
```

Comment:
You probably should make the PHP 5.2 specific check after the `||` more precise:
1. This only occurs when the token is of the type `T_INLINE_HTML`.
2. I think the `trim()` should probably be removed as it could cause false positives.
3. The next token `$stackPtr +1` may not exist and could throw an `undefined index` notice now.
4. The next token `$stackPtr +1` should also be of the type `T_INLINE_HTML`.